### PR TITLE
3677

### PR DESCRIPTION
**Pull request by hrizvi1**
This for work item https://github.ibm.com/cp4i/cp4i-level3/issues/26
and https://github.ibm.com/cp4i/icip-issues/issues/2200

The successful run from 1-click via global catalog:  [link](https://cloud.ibm.com/schematics/workspaces/us-south.workspace.ibm-cp-integration-04-13-2022-3677-7.09313fda/log/2f858b8df40ac7122141359d448b786c)

image

1-click automation run that will test launch via script directly: https://hyc-cip-jenkins.swg-devops.com/blue/organizations/jenkins/Automation%2Fdev%2F1-click-automation/detail/PR-986/7/pipeline/28

image

image

test changes made to the automation build to test the changes:
https://github.ibm.com/cp4i/icp4i-pipeline/pull/986/files - these changes will not be merged these were made for testing individual parameters 

```
I used this setting for the above runs
{
  "cognitiveCarRepair": "true",
  "drivewayDentDeletion": "false",
  "eventEnabledInsurance": "false",
  "mappingAssist": "true",
  "weatherChatbot": "true"
}

```

Also ran another run on 1-click with different combo of values: https://cloud.ibm.com/schematics/workspaces/us-south.workspace.ibm-cp-integration-04-13-2022-9.99f830e6/log/ca6d2fc0472eb37304b836ea836dcfd9

it failed b/c of eei pre-reqs scripts failed

image

### CODE REVIEW

**Comment by hrizvi1** [comment]
@DanRoseus I printed out the contents of REQUIRED_DEMOS_JSON:

image

and all demos are one json object instead of each demo being an object

**Comment by hrizvi1** [comment]
Discussion about line 403 here: https://ibm-cloud.slack.com/archives/C016VR7HK42/p1649923547498839

File: products/bash/1-click-install.sh [hunk 1]
```diff
@@ -484,13 +517,16 @@ else
...
+if [[ "$demoPreparation" == "true" || "$drivewayDentDeletionDemo" == "true" || "$eventEnabledInsuranceDemo" == "true" || "$cognitiveCarRepairDemo" == "true" || "$mappingAssistDemo" == "true" || "$weatherChatbotDemo" == "true" ]]; then
   divider && echo -e "$INFO [INFO] Setting up all required addons, products and demos in the '$JOB_NAMESPACE' namespace..."
   CURRENT_DIR_WITHOUT_DOT_SLASH=${CURRENT_DIR//.\//}
 
   # create a new backup json file to revert demos.json after setup script
   cp $CURRENT_DIR_WITHOUT_DOT_SLASH/demos.json $CURRENT_DIR_WITHOUT_DOT_SLASH/demos-backup.json
 
+  echo -e "Printing the env vars"
+  env
+
```
Comment (DanRoseus) [inline products/bash/1-click-install.sh:529]: Don't think this is needed now?

**Review by DanRoseus — CHANGES_REQUESTED** [review]
One minor comment...

**Review by DanRoseus — APPROVED** [review]
Looks good!

**Pull request merged**